sv/git.go

```diff
@@ -194,23 +194,33 @@ func parseLogOutput(messageProcessor MessageProcessor, log string) []GitCommitLo
 	var logs []GitCommitLog
 	for scanner.Scan() {
 		if text := strings.TrimSpace(strings.Trim(scanner.Text(), "\"")); text != "" {
-			logs = append(logs, parseCommitLog(messageProcessor, text))
+			log, err := parseCommitLog(messageProcessor, text)
+			// Ignore errors occuring during parsing
+			if err == nil {
```

Comment:
i think it's better to return an error in this case.. if by any means the regex is invalid, it will return an empty log and it would be difficult to know why..

what we could do is change the `prepareHeader` to return the original message if the regex is not found (i think this is what you were talking about having a must-have or not).. or just keep as it is and expect that the regex will handle the prefix as optional..

Reply:
for example:

for a gitlog with the headers above:

```
feat: add a feature
Merged PR 123: feat: something
```

the regex in the config should be: `(Merged PR (\\d+): )?(?P<header>.*)`

Reply:
maybe this way (without changing the `prepareHeader`) is better so the person have the right to choose the wanted behavior in the regex

Reply:
I'll introduce a switch to ignore unparsable commits

Reply:
Is implemented.

Reply:
I think that just returning the error is enough.. sorry I probably wrote this in a confusing way..

if we keep like this, the regex `(Merged PR (\\d+): )?(?P<header>.*)` would work for both scenarios, and if it was wrong it would fail the execution returning an error..

```go
func (g GitImpl) parseLogOutput(messageProcessor MessageProcessor, log string) ([]GitCommitLog, error) {
	scanner := bufio.NewScanner(strings.NewReader(log))
	scanner.Split(splitAt([]byte(endLine)))
	var logs []GitCommitLog
	for scanner.Scan() {
		if text := strings.TrimSpace(strings.Trim(scanner.Text(), "\"")); text != "" {
			log, err := parseCommitLog(messageProcessor, text)
			if err != nil {
				return nil, err
			}
			logs = append(logs, log)
		}
	}
	return logs, nil
}
```

Reply:
Removed the config option :)

Reply:
thanks a lot :heart: